stdlib/pqueue.mli

```diff
@@ -0,0 +1,299 @@
+
+    val add: t -> elt -> unit
+    (** [add q x] adds the element [x] in the priority queue [q]. *)
+
+    val add_seq: t -> elt Seq.t -> unit
+    (** [add q s] adds the elements of [s] in the priority queue [q]. *)
+
+    exception Empty
+    (** Raised when {!min_elt}, {!pop_min} or {!remove_min} is applied
+        to an empty queue. *)
+
+    val min_elt: t -> elt
+    (** [min_elt q] returns an element in queue [q] with minimal priority,
+        without removing it from the queue,
+        or raises {!Empty} if the queue is empty. *)
+
+    val min_elt_opt: t -> elt option
+    (** [min_elt_opt q] returns an element in queue [q] with minimal priority,
+        without removing it from the queue, or returns [None] if the queue
+        is empty. *)
+
+    val pop_min: t -> elt
+    (** [pop_min q] removes and returns an element in queue [q] with
+        minimal priority, or raises {!Empty} if the queue is empty. *)
+
+    val pop_min_opt: t -> elt option
+    (** [pop_min_opt q] removes and returns an element in queue [q] with
+        minimal priority, or returns [None] if the queue is empty. *)
+
+    val remove_min: t -> unit
+    (** [remove_min q] removes an element in queue [q] with
+        minimal priority, or raises {!Empty} if the queue is empty. *)
+
+    val clear: t -> unit
+    (** Discard all elements from a priority queue. *)
+
+    val copy: t -> t
+    (** [copy q] is a shallow copy of [q] (a new priority queue
+        containing the same elements as [q]). *)
+
+    (** {1:conversions Conversions from other data structures} *)
+
+    val of_array: elt array -> t
+    (** [of_array a] returns a new priority queue containing the
+        elements of array [a]. Runs in linear time. *)
+
+    val of_list: elt list -> t
+    (** [of_list l] returns a new priority queue containing the
+        elements of list [l]. Runs in linear time. *)
+
+    val of_seq: elt Seq.t -> t
+    (** [of_seq s] returns a new priority queue containing the
+        elements of sequence [s]. Runs in linear time. *)
+
+    (** {1:iteration Iteration}
+
+        The order in which the elements of a priority queue are
+        traversed is unspecified.
+
+        It is a programming error to mutate a priority queue (by
+        adding or removing elements) during an iteration of the queue.
+        Such an error may be detected and signaled by the backing dynamic
+        array implementation, but this is not guaranteed. *)
+
+    val iter: (elt -> unit) -> t -> unit
+    (** [iter f q] calls [f] on each element of [q] in some
+        unspecified order. *)
+
+    val fold: ('acc -> elt -> 'acc) -> 'acc -> t -> 'acc
+    (** [fold f accu q] is [(f (... (f (f accu x1) x2) ...) xn)]
+        where [x1,x2,...,xn] are the elements of [q] in some
+        unspecified order. *)
+
+    val to_seq: t -> elt Seq.t
+    (** [to_seq q] is the sequence of elements of [q] in some
+        unspecified order. *)
+
```

Comment:
Not necessarily asking for but it. But should we add the usual `equal` and (priority aware) `compare` functions ?